src/main/java/nl/knaw/dans/pf/language/ddm/handlers/DcLanguageHandler.java

```diff
@@ -25,9 +25,28 @@
     protected void finishElement(final String uri, final String localName) throws SAXException {
         BasicString basicString = createBasicString(uri, localName);
         if (basicString != null) {
-            basicString.setScheme("ISO 639");
-            basicString.setSchemeId("common.dc.language");
+            String emdLanguageCode = getEmdLanguageCode(basicString.getValue());
+            if (emdLanguageCode != null) {
+                basicString.setValue(emdLanguageCode);
+                basicString.setScheme("ISO 639");
+                basicString.setSchemeId("common.dc.language");
+            }
             getTarget().getEmdLanguage().getDcLanguage().add(basicString);
         }
     }
+
+    private String getEmdLanguageCode(final String language) {
```

Comment:
I see what you did here, but isn't there an external file with this mapping that can be used instead? I mean, this is all hardcoded stuff...